src/fsharp/vs/service.fs

```diff
@@ -2854,22 +2817,37 @@ type FSharpChecker(legacyReferenceResolver, projectCacheSize, keepAssemblyConten
 
     member ic.ReferenceResolver = legacyReferenceResolver
 
-    member ic.MatchBraces(filename, source, options, ?userOpName: string) =
+    member ic.MatchBraces(filename, source, options: FSharpParsingOptions, ?userOpName: string) =
         let userOpName = defaultArg userOpName "Unknown"
-        async { 
-            match braceMatchCache.TryGet (AssumeAnyCallerThreadWithoutEvidence(), (filename, source, options)) with 
+        async {
+            match braceMatchCache.TryGet(AssumeAnyCallerThreadWithoutEvidence(), (filename, source, options)) with
             | Some res -> return res
-            | None -> 
-                let! res = backgroundCompiler.MatchBraces(filename, source, options, userOpName)
-                braceMatchCache.Set (AssumeAnyCallerThreadWithoutEvidence(), (filename, source, options), res)
-                return res 
-         }
+            | None ->
+                let res = Parser.matchBraces(source, filename, options, userOpName)
+                braceMatchCache.Set(AssumeAnyCallerThreadWithoutEvidence(), (filename, source, options), res)
+                return res
+        }
 
-    member ic.ParseFileInProject(filename, source, options, ?userOpName: string) =
+    member ic.GetParsingOptionsFromProjectOptions(options): FSharpParsingOptions =
+        let sourceFiles = List.ofArray options.SourceFiles
+        let argv = List.ofArray options.OtherOptions
+        let parsingOptions, _ = ic.GetParsingOptionsFromSourceFilesAndCommandLineArgs(sourceFiles, argv)
+        parsingOptions
+
+    member ic.MatchBraces(filename, source, options: FSharpProjectOptions, ?userOpName: string) =
+        let userOpName = defaultArg userOpName "Unknown"
+        ic.MatchBraces(filename, source, ic.GetParsingOptionsFromProjectOptions(options), userOpName)
+
+    member ic.ParseFile(filename, source, options, ?userOpName: string) =
         let userOpName = defaultArg userOpName "Unknown"
         ic.CheckMaxMemoryReached()
-        backgroundCompiler.ParseFileInProject(filename, source, options, userOpName)
-        
+        backgroundCompiler.ParseFile(filename, source, options, userOpName)
+
+
+    member ic.ParseFileInProject(filename, source, options, ?userOpName: string) =
+        let userOpName = defaultArg userOpName "Unknown"
+        ic.ParseFile(filename, source, ic.GetParsingOptionsFromProjectOptions(options), userOpName)
```

Comment:
Now I see the implementation of ``ParseFileInProject`` I'm concerned it's a bit costly, involving the full reparse of the command line arguments, setting up a TcConfigBuilder and so on.   Hmmm.... I suppose we should just deprecate this and adjust the Visual F# Tools to use ``ParseFile``

I will aim to take a look at that tomorrow.